packages/table/src/interactions/reorderable.tsx

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+import { Utils } from "../common/utils";
+import { Draggable, ICoordinateData, IDraggableProps } from "../interactions/draggable";
+import { IRegion, RegionCardinality, Regions } from "../regions";
+
+export interface IReorderableProps {
+    /**
+     * A callback that is called while the user is dragging to reorder.
+     *
+     * @param oldIndex the original index of the element or set of elements
+     * @param newIndex the new index of the element or set of elements
+     * @param length the number of contiguous elements that were moved
+     */
+    onReorderPreview: (oldIndex: number, newIndex: number, length: number) => void;
```

Comment:
i think i called this onResizing for the resize case. Either way, would be nice to be consistent